etl-routes.js

```diff
@@ -692,38 +695,50 @@ module.exports = function () {
                     },
                     handler: function (request, reply) {
                         let EIDLabReminderService = require('./service/eid/eid-lab-reminder.service');
-                        EIDLabReminderService.pendingEIDReminders(request.params, config.hivLabSystem)
-                            .then((eidReminders) => {
-                                let combineRequestParams = Object.assign({}, request.query, request.params);
-                                combineRequestParams.limitParam = 1;
-                                let reportParams = etlHelpers.getReportParams('clinical-reminder-report', ['referenceDate', 'patientUuid', 'offSetParam', 'limitParam'], combineRequestParams);
-
-                                let report = new BaseMysqlReport('clinicalReminderReport', reportParams.requestParams);
-                                report.generateReport().then((results) => {
-                                    try {
-                                        if (results.results.results.length > 0) {
-                                            let processedResults = patientReminderService.generateReminders(results.results.results, eidReminders);
-                                            results.result = processedResults;
-                                        } else {
-                                            results.result = {
-                                                person_uuid: combineRequestParams.person_uuid,
-                                                reminders: []
-                                            };
-                                        }
-                                        reply(results);
-                                    } catch (error) {
-                                        console.log('Error generating reminders', error);
-                                        reply(Boom.badImplementation('An internal error occurred'));
-                                    }
+                        let combineRequestParams = Object.assign({}, request.query, request.params);
+                        syncPreproc.processPendingLabResultRequest(request.params.patientUuid)
```

Comment:
The line 698 to 732 is really business logic, and having it placed here at the 'controller' starts to  complicate the controller making it hard to maintain in the long-run. etl-routes.js file is already too big at this point.